test/e2e/utils/test_context.go

```diff
@@ -246,6 +246,29 @@ func (t *TestContext) Destroy() {
 	}
 }
 
+// DestroyOutputDir is written separately to ensure the deletion
+// of the output directory in case of scaffolded projects
+func (t *TestContext) DestroyOutputDir() {
```

Comment:
We have already: https://github.com/kubernetes-sigs/kubebuilder/blob/de4c4c719c295513ffb824d71469a699f0b86150/test/e2e/utils/test_context.go#L229-L247

Could we not use what we have?
Then, due to the "output" dir being specific for those tests, do we call the removal on the tests specifically?